DOC: Update examples

bocklund opened the pull request:
* Binary examples updated
  - Fix bug in `ax=ax` and `tielines=False` for the newer `binplot` API.
  - More details at the top about where to get TDBs and how to make sure it's in the right directory for a user's calculation
  - Make all the binary examples (phase diagrams and energies) self contained in each cell for easier copy-pasting
  - Comment all the binary examples so copy-pasters have some explanation
  - Add Al-Mg example, since it's a more complicated diagram than Al-Zn, but is still relatively fast (~40s on my machine). Contrast this with the other non-trivial examples (Al-Ni, Al-Fe, Nb-Re) which are all relatively slow.
  - Add citations for papers that exist with assessments.
  - Add axes labels to Gibbs energy curves example
* Add example for calculating and plotting along a specific internal degree of freedom

bocklund commented:
This is ready for review.

See the two updated docs here:
https://github.com/pycalphad/pycalphad/blob/doc-update-examples/examples/BinaryExamples.ipynb
https://github.com/pycalphad/pycalphad/blob/doc-update-examples/examples/ReferenceStateExamples.ipynb (only change is the new last cell and text)

richardotis commented:
Minor things:

1. > The Thermodynamic DataBase DataBase (TDBDB) has index many available databases

indexed

2. We still have a hard time with the really flat liquidus in the Nb-Re diagram. Not really in scope for this PR, but an observation.
3. "Enthalpy of mixing" first example
```python
ax.scatter(result.X.sel(component='RE'), result.HMR, marker='.', s=5, label='CHI_RENB')
ax.set_xlim((0, 1))
ax.set_xlabel('X(RE)')
ax.set_ylabel('HM_MIX')
ax.set_title('Nb-Re CHI Mixing Enthalpy')
```
I think both of these should be labeled liquid, not chi.

4. The convex hull example is really cool because it shows off the power of the multi-dimensional representation of the calculation results.
5. Last reference state example: `result.Y.squeeze()[:, 2]` - for a new person this syntax is confusing. A portion of the text explanation at the top (about the site fraction ordering) should be copied into a comment in the code example. Then something like `Y_CHI_1_RE = result.Y.squeeze()[:,2]` before using the new array in the plot.

bocklund commented:
1. Fixed
2. Yeah. I think it could be improved for that particular diagram by increasing the point density when calculating the grid, so that the two phase regions would be picked up by the hyperplane. Not sure whether that would take away from the example (or if that example should be removed or moved somewhere else with those "advanced" API usage). A line following mapping approach might be a longer term fix.
3. Fixed
4. Thanks, I agree! I think whatever the solution for #154 looks like will help to make these even easier to create
5. Comments added and `Y_CHI_1_RE` factored out.
